lint-staged.config.js

```diff
@@ -11,10 +11,10 @@ const nonEslintPrettierExtensions = prettierExtensions.filter(ext => !eslintExte
 // https://www.npmjs.com/package/lint-staged
 module.exports = {
   // Run eslint in fix mode followed by prettier
-  [`*.{${eslintExtensions.join(',')}}`]: ['node ./scripts/lint-staged/eslint', 'prettier --write'],
+  [`*.{${eslintExtensions.join(',')}}`]: ['node ./scripts/lint-staged/eslint', 'node ./scripts/format.js'],
```

Comment:
I don't think this is good idea. I'd recommend to get rid of that custom `format` script completely and use raw prettier. approach of running checks only on affected packages should be applied in general not just for prettier. there are excellent oss tools that will help us with that.

Reply:
I agree with using raw prettier, but objective of this PR is to align the two checks. This is no longer a huge issue since we have single version policy on prettier (great !). 

Before the single version policy `format` and precommit actually ran v1 and v2 versions of prettier separately.

Do think that there is no more reason for the changes in this PR ?

Reply:
My POV is that even if we're using custom stuff then we should at least be consistent

Reply:
I cant agree more regarding consistency. in this case that script is doing unnecessary work on local machine. and as you pointed out, issues with various prettier versions is no longer present.

to wrap it up: your PR was already accepted by owner of this codebase so its your call now :)

Reply:
On second thought I agree with @Hotell on using raw prettier where possible. This change actually turned out to break lint-staged since it doesn't respect the list of filenames. So I made a PR #17281 reverting to using prettier for lint-staged, and adding clarification that `format` is only intended for a couple special cases.